src/itnovum/openITCOCKPIT/ConfigGenerator/PhpFpmOitc.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+
+namespace itnovum\openITCOCKPIT\ConfigGenerator;
+
+
+use Cake\Core\Configure;
+
+class PhpFpmOitc extends ConfigGenerator implements ConfigInterface {
+    /** @var string */
+    protected $templateDir = 'PhpFpmOitc';
+    /** @var string */
+    protected $template = 'oitc.conf.tpl';
+    /** @var string */
+    protected $realOutfile = '';
+    /** @var string */
+    protected $linkedOutfile = '';
+    /** @var string */
+    protected $commentChar = ';';
+    /** @var array */
+    protected $defaults = [
+        'int' => [
+            'max_children' => 5,
+        ],
+    ];
+
+    /**
+     * PhpFpmOitc constructor.
+     * determines the current php version as the file gets written in the latest php-fpm pool.d directory
+     * this changes with every php update (7.4 to 7.5 to 7.6 and so on)
+     */
+    public function __construct() {
+        try {
+            $version = phpversion();
+            preg_match('/[^.]*.[^.]*/', $version, $match);
+            if (is_array($match)) {
+                $version = $match[0];
+                if (file_exists('/etc/php/' . $version)) {
+                    $this->realOutfile = '/etc/php/' . $version . '/fpm/pool.d/oitc.conf';
+                    $this->linkedOutfile = '/etc/php/' . $version . '/fpm/pool.d/oitc.conf';
+                }else{
+                    throw new \Exception('/etc/php/' . $version. ' directory does not exists');
+                }
```

Comment:
Please use `substr(PHP_VERSION, 0, 3)` to get the current PHP Version (same method as UPDATE.sh and SETUP.sh) uses, so if it fails it fails the same way on all scripts.

Also use `is_dir` instead of `file_exists`